File: 백준/Bronze/4153. 직각삼각형/직각삼각형.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
public class Main {
    public static void main(String[] args) throws NumberFormatException, IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        while(true){
            String [] str = br.readLine().split(" ");
            int [] triangle = new int[3];
            for(int i=0;i<str.length;i++){
                triangle[i]=Integer.parseInt(str[i]);
            }
            if(triangle[0]==0 && triangle[0]==0 && triangle[0]==0){
                break;
            }else{
                Arrays.sort(triangle);
                if(triangle[triangle.length-1]*triangle[triangle.length-1]
                ==triangle[0]*triangle[0]+triangle[1]*triangle[1]){
                    System.out.println("right");
                }else{
                    System.out.println("wrong");
                }
            }


        }
        
            
    }
}

File: 백준/Bronze/4153. 직각삼각형/직각삼각형.swift
//
//  main.swift
//  BJ4153
//
//  Created by 안홍범 on 9/4/24.
//

import Foundation

while true {
    let input = readLine()!.split(separator: " ").map{Int($0)!}
    
    if input[0] == 0 && input[1] == 0 && input[2] == 0 {
        break
    }
    
    let sortedSides = input.sorted()
    let a = sortedSides[0]
    let b = sortedSides[1]
    let c = sortedSides[2]
    
    if a * a + b * b == c * c {
        print("right")
    } else {
        print("wrong")
    }
}
